### PR TITLE
feat: refactoring the settings page in the dashboard

### PR DESCRIPTION
**Pull request by Ebubeker**
Did a lot of changes, I created the custom hooks just to make it a bit easier to read the code, rather than having useEffects inside the components. I also removed the static data, except for the column names. Did also some changes in the home page regarding the review done by Ana

### CODE REVIEW

File: app/client-v2/packages/ui/components/DataDisplay/TreeViewWithCheckboxes.tsx [hunk 1]
```diff
@@ -7,6 +9,7 @@ import React, { useState, useEffect } from "react";
 
 import { makeStyles } from "../../lib/ThemeProvider";
 import { Text } from "../theme";
+import { Tooltip } from "./Tooltip";
```
Comment (baghdasarovnorayr) [inline app/client-v2/packages/ui/components/DataDisplay/TreeViewWithCheckboxes.tsx:12]: please use absolute paths

File: app/client-v2/apps/goat/hooks/dashboard/TeamsHooks.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from "react";
+import { Team } from "../../app/(dashboard)/settings/organization/Teams";
```
Comment (baghdasarovnorayr) [inline app/client-v2/apps/goat/hooks/dashboard/TeamsHooks.ts:2]: please use absolute paths

**Review by majkshkurti — APPROVED** [review]
Thanks.

File: app/client-v2/apps/goat/hooks/dashboard/TeamsHooks.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    }
+  }
+
+  return { saveEditTeam };
+}
+
+export function useTeamSearch(rawRows: Team[], searchText: string | undefined) {
+  const [rows, setRows] = useState<Team[]>([]);
+
+  useEffect(() => {
```
Comment (baghdasarovnorayr) [inline app/client-v2/apps/goat/hooks/dashboard/TeamsHooks.ts:68]: you dont need useEffect, if props are changed component will rerender.

`return filterSearch(rawRows, "name", searchText || "")`

File: app/client-v2/apps/goat/hooks/dashboard/OrganisationHooks.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import { filterSearch, makeArrayUnique } from "@/lib/utils/helpers";
+import type { User } from "manage-users-dashboard";
+
+// Get all users in order to manage them
+export const useUsersData = (searchWord?: string) => {
+  const UsersFetcher = (url: string) => {
+    return axios(url).then((res) => res.data);
+  };
+
+  const { data, error, isLoading } = useSWR("/api/dashboard/organization", UsersFetcher);
```
Comment (baghdasarovnorayr) [inline app/client-v2/apps/goat/hooks/dashboard/OrganisationHooks.ts:13]: I suggest keep keys ("/api/dashboard/organization") in the constant's

File: app/client-v2/apps/goat/app/(dashboard)/settings/organization/layout.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      name: "Teams",
+    },
+    {
+      child: <div>overview</div>,
+      name: "Settings",
+    },
+  ];
+
+  return (
+    <>
```
Comment (baghdasarovnorayr) [inline app/client-v2/apps/goat/app/(dashboard)/settings/organization/layout.tsx:28]: This fragment  is no nessary

**Review by baghdasarovnorayr — CHANGES_REQUESTED** [review]

**Pull request merged**